test/externalTests/zeppelin.sh

```diff
@@ -42,8 +42,9 @@ function test_fn { npm test; }
 function zeppelin_test
 {
     local repo="https://github.com/OpenZeppelin/openzeppelin-contracts.git"
-    local ref_type=branch
-    local ref="master"
+    #TODO Revert back to master when https://github.com/OpenZeppelin/openzeppelin-contracts/issues/4349 is fixed.
+    local ref_type=commit
+    local ref="002a7c8812e73c282b91e14541ce9b93a6de1172"
```

Comment:
```suggestion
    local ref_type=tag
    local ref=v4.9.2
```

Would this work? Or perhaps `v4.9.1`? It would be best if we could use tags.

Reply:
`4.9.2` does not work - already tried it - it fails with different test cases. I haven't tried `4.9.1`, but I'm curious as to why that's better than using a specific hash - ultimately, if it isn't `master`, it's the same thing, no?

Reply:
Tags are more descriptive so I'd always take one over a hash that tells me nothing at a glance. You can check but it's a pain to do. I mean, you could add a comment saying where the commit points to, but you did not do that here, which exemplifies my point :) I guess it's the merge point of https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4362, which is somewhere after `v4.9.2`?

I'd also tend to stick to releases when we can because this way, if we need to reproduce some problem outside of our CI, we can easily install that version from any other source where releases are available, like npm. This matters more for things that require some complex (or just long and annoying) setup, but I don't see why we would close off that option if we don't have to.

In the end, a commit is fine, just not the best option IMO. `v4.9.1` is just 2 weeks old so if that works, it would be my first choice. If it doesn't, we can take that commit, just please add a comment.

Reply:
I'll re-run it locally with the `4.9.1` tag and will update the PR in case it passes.

Reply:
By the way, an even better option might be this: https://github.com/ethereum/solidity/pull/14295/commits/2a5f046e87d7c3a13c462c6eed7773f8920d8d26. Looks like the contributor fixed the breakage. We could just cherry pick that and have OZ working without pinning anything.

Reply:
@cameel `4.9.1` doesn't work either. I've left the commit hash as is, and added another comment. Please take a look.

Reply:
Have you tried cherry-picking that commit from #14295 I mentioned?

Reply:
> By the way, an even better option might be this: [2a5f046](https://github.com/ethereum/solidity/commit/2a5f046e87d7c3a13c462c6eed7773f8920d8d26) (from #14295). Looks like the contributor fixed the breakage. We could just cherry pick that and have OZ working without pinning anything.

I mean, that's always an option for any failing test, but how often are we gonna have to do this with external tests? Especially for OZ, which is a required step.

Reply:
But in this case we have it already done so why not take it.

I mean, if pinning would guarantee no breakage, that would be the better choice, but I'm assuming that it will break sooner or later and we'll have to update anyway.